docs/SETUP_TLS_CERTIFICATES.md

```diff
@@ -0,0 +1,232 @@
+# Setting up TLS certificates for secure plugin communication in Snap
+
+<!-- TOC -->
+
+- [Setting up TLS certificates for secure plugin communication in Snap](#setting-up-tls-certificates-for-secure-plugin-communication-in-snap)
+    - [Overview](#overview)
+        - [Remarks](#remarks)
+    - [Obtaining self-signed certificates](#obtaining-self-signed-certificates)
+        - [Remarks](#remarks-1)
+        - [Procedure](#procedure)
+    - [Obtaining legitimate certificate from a public CA Let's Encrypt](#obtaining-legitimate-certificate-from-a-public-ca-lets-encrypt)
+        - [Prerequisites](#prerequisites)
+        - [Remarks](#remarks-2)
+        - [Procedure](#procedure-1)
+    - [Installing certificates in system (Ubuntu, CentOS)](#installing-certificates-in-system-ubuntu-centos)
+        - [Installing root certificate in Ubuntu](#installing-root-certificate-in-ubuntu)
+        - [Installing root certificate in Centos](#installing-root-certificate-in-centos)
+    - [Testing TLS-secured plugin communication in local setup](#testing-tls-secured-plugin-communication-in-local-setup)
+        - [Starting a telemetry workflow:](#starting-a-telemetry-workflow)
+
+<!-- /TOC -->
+
+
+## Overview
+
+This document provides instructions on how to obtain certficates to evaluate TLS-secured plugin communication in Snap.
+
+Following routines are described:
+- obtaining self-signed certificates for evaluation in local test environment,
+- obtaining legitimate certificate from a certain CA authority,
+- installing root certificate in OS for validating endpoints' certificates,
+- starting telemetry workflow with secured plugin communication.
+
+### Remarks
+- This document can't be considered a definite guide to obtaining certificates. It aims only as guiding user in acquiring a certificate.
+- This guide doesn't indicate how to handle certificate revocation as that service is not yet supported in Snap.
+- Every organization has to review its requirements prior to obtaining certificates or setting up any secure communication scheme.
+
+## Obtaining self-signed certificates
+
+This section describes how to obtain certificates signed by a local CA.
+
+In the course of this guide user will establish a local CA, becoming able to establish local chains of trust. Thus the user will be able to generate certificates for Snap plugins and Snap daemon (`snapteld`) that all share common trust root.
+
+### Remarks
+
+- CA private key needs to be protected, possibly on a dedicated machine or media.
```

Comment:
Remote period at end of bullet. 

Suggestion (not a requirement): s/needs to be/should be/